src/Tasks/Microsoft.NET.Build.Tasks/ResourceUpdater.cs

```diff
@@ -169,6 +172,47 @@ protected override bool ReleaseHandle()
         /// </summary>
         private readonly SafeUpdateHandle hUpdate;
 
+        ///<summary>
+        /// Determines if the ResourceUpdater is supported by the current operating system.
+        /// Some versions of Windows, such as Nano Server, do not support the needed APIs.
+        /// </summary>
+        public static bool IsSupportedOS()
+        {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return false;
+            }
+
+            var module = Kernel32.LoadLibraryEx("kernel32.dll", IntPtr.Zero, 0);
+            if (module == IntPtr.Zero)
+            {
+                return false;
+            }
+
+            // Ensure the export exists before P/Invoking
+            var proc = Kernel32.GetProcAddress(module, "BeginUpdateResourceW");
+            Kernel32.FreeLibrary(module);
+            if (proc == UIntPtr.Zero)
+            {
+                return false;
```

Comment:
I'm fine with that approach if others prefer it.  It would simplify this code a little and it should only occur once a build on (presumably) only Nano Server.

Reply:
Catching the exception is certainly easier / less code - if the noise is acceptable.

Reply:
Also, if this is a temporary fix (i.e. if we intend to rip this fix out for a cross-platform resource updater), then I'm more inclined to just catch the `EntryPointNotFoundException`, which seems to only occur for Windows Server 2016 Nano Server (end-of-support, I think).